### DISCUSSION TITLE
Color Change Transient Prompt

### DISCUSSION THREAD

**Discussion by jlwoolf** [JanDeDobbeleer/oh-my-posh#1964] [Troubleshoot]
Is there a way to configure the transient prompt such that the color of the foreground is different depending on the commands output? The .Code variable is only available in the exit segment. Like being able to use $? from the shell?

```
"transient_prompt": {
    "background": "transparent",
    "foreground": "green",
    "template": "{{ if $? }}<red,transparent>\u276f</>{{ else }}<green,transparent>\u276f</>{{ end }} "
  }
```

**Comment by lewis-yeung** [comment]
Obviously `$?` will not work since it is not an available identifier in Go template actions. Would it be a good idea to make `.Code` a property of template in the transient prompt config in a later release?

**Reply by JanDeDobbeleer** [reply]
I'll need to make that variable globally available, but yes, can do.

**Reply by JanDeDobbeleer** [reply]
Incoming together with color templates. Just need to fix my errors and validate everything.

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung this is [available](https://ohmyposh.dev/docs/config-transient) now.
